algorithms.py

```diff
@@ -281,6 +282,221 @@ def find_indv(self, indv):
         return self._ancestors[indv.label].index(indv)
 
 
+class PyPedigree(object):
```

Comment:
True it's not really necessary, I was just separating the methods which are implemented in C or Python. Figured it could be useful down the line for working out tricky stuff like multiple pops etc., but it doesn't do much now. Better to merge them you think?

Reply:
Well either merge or rename. Or, we could use the Pedigree class from the main msprime code to do the file reading etc. There's no much point in having an exact duplicate of the same thing here.

Reply:
That's a good point. Updated to use `msprime.Pedigree`, and a few other quick updates to better match current C implementation.